os/include/tinyara/sched.h

```diff
@@ -615,9 +618,10 @@ struct tcb_s {
 	uint32_t ram_start;		/* Start address of RAM partition for this app */
 	uint32_t ram_size;		/* Size of RAM partition for this app */
 	uint32_t uspace;		/* User space object for app binary */
+	uint32_t uheap;			/* User heap object pointer */
 
 #ifdef CONFIG_ARMV7M_MPU
-	uint32_t mpu_regs[MPU_REGS];
+	uint32_t mpu_regs[3 * MPU_NUM_REGIONS];	/* We need 3 register values to configure each MPU region */
```

Comment:
If CONFIG_OPTIMIZE_APP_RELOAD_TIME is enabled, this is correct, but CONFIG_OPTIMIZE_APP_RELOAD_TIME is not set, we needs different size.

Reply:
I fixed it by adjusting the value of MPU_NUM_REGIONS based on CONFIG_OPTIMIZE_APP_RELOAD_TIME

Reply:
Yes. We need 3 registers for each memory region. RNR. RABR and RSAR.
I have mentioned the same in the comments.